api/v1alpha1/envoygateway_types.go

```diff
@@ -364,10 +364,10 @@ type RateLimitTracing struct {
 	// BackendRef defines the target trace collector endpoint configuration
 	BackendRef gwapiv1.BackendObjectReference `json:"backendRef"`
 
-	// Protocol defines the protocol of provider in tracing feature.
+	// AppProtocol defines the protocol of provider port in tracing feature.
 	// Only "http"(default) and "grpc" are allowed in this field
 	// +optional
-	Protocol string `json:"protocol,omitempty"`
+	AppProtocol *string `json:"appProtocol"`
```

Comment:
I think you can read this from corev1.Service?

Reply:
Since we currently do not plan to use k8s client to obtain Service in infra, but use fqdn, so we have no way to read this from Service, people still need to actively set it.

Reply:
so the plan is use something like `[name].[namespace].svc:[port]`, even type is ExternalName?
if so, can you address how to use this in API or doc?

Reply:
Currently I have no plans to support ExternalName, I thought I would add to the documentation and explain this in the documentation.  Thank you for your review :)

Reply:
yeah lets not support `ExternalName` 

I think what is needed for this is to make sure 
* we can reconcile this backendRef In the provider, similar to https://github.com/envoyproxy/gateway/blob/deea89571d227647669d8361f7c8399c8f0144bd/internal/provider/kubernetes/controller.go#L465
* pass it over to gateway-api via Resources
* convert to Infra IR with a new section for RL https://github.com/envoyproxy/gateway/blob/main/internal/ir/infra.go
* populate in infra mgr


I keep going back and forth as to whether this is overkill or not, or should we support explicit fields for this field (RateLimit.Telemetry.Tracing)

Pros of `backendRef` approach
* Leverage fields within `Service`
* Reuse BackendTLSPolicy to decide on whether to connect over TLS or not
* Define it once, and can be used multiple times

Cons
* More involved code

I'm still inclined to the `backendRef` approach

Reply:
+1